src/mono/browser/browser.proj

```diff
@@ -63,6 +63,7 @@
       <WasmPInvokeModule Include="libSystem.Native" />
       <WasmPInvokeModule Include="libSystem.IO.Compression.Native" />
       <WasmPInvokeModule Include="libSystem.Globalization.Native" />
+      <WasmPInvokeModule Include="libz" />
```

Comment:
@pavelsavara @ilonatommy @kg I wonder if this is what's causing all the problems. I added this without any conditions.  

Anyway, since now I'm working on including wasm among the platforms that will keep using system zlib, this line should go away.